Sources/Public/CalendarItemView.swift

```diff
@@ -0,0 +1,67 @@
+// Created by Bryan Keller on 7/15/20.
+// Copyright © 2020 Airbnb Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+// MARK: - CalendarItemView
+
+/// A protocol to which all views displayed in `CalendarView` must conform. By conforming to this protocol, `CalendarView` is
+/// able to treat each view as a function of its `initialConfiguration` and its `viewModel`, making the declarative API much
+/// easier to reason about.
+public protocol CalendarItemView: UIView {
```

Comment:
@brynbodayle here's the new protocol that all views in the calendar need to conform to. This, along with the changes in `CalendarItem.swift`, are a good starting place for understanding how this change works.